core/src/main/java/org/elasticsearch/action/ingest/SimulateExecutionService.java

```diff
@@ -65,12 +66,15 @@ SimulateDocumentResult executeDocument(Pipeline pipeline, IngestDocument ingestD
     }
 
     public void execute(SimulatePipelineRequest.Parsed request, ActionListener<SimulatePipelineResponse> listener) {
-        threadPool.executor(THREAD_POOL_NAME).execute(() -> {
-            List<SimulateDocumentResult> responses = new ArrayList<>();
-            for (IngestDocument ingestDocument : request.getDocuments()) {
-                responses.add(executeDocument(request.getPipeline(), ingestDocument, request.isVerbose()));
+        threadPool.executor(THREAD_POOL_NAME).execute(new ActionRunnable<SimulatePipelineResponse>(listener) {
+            @Override
+            protected void doRun() throws Exception {
+                List<SimulateDocumentResult> responses = new ArrayList<>();
+                for (IngestDocument ingestDocument : request.getDocuments()) {
+                    responses.add(executeDocument(request.getPipeline(), ingestDocument, request.isVerbose()));
+                }
+                listener.onResponse(new SimulatePipelineResponse(request.getPipeline().getId(), request.isVerbose(), responses));
```

Comment:
if you just use a runnable and the execution fails or the task gets rejected we never call the listener?